crates/rpc-types-mev/src/eth_calls.rs

```diff
@@ -311,6 +311,34 @@ pub struct PrivateTransactionRequest {
     pub preferences: PrivateTransactionPreferences,
 }
 
+impl<T: Encodable2718> From<T> for PrivateTransactionRequest {
+    fn from(envelope: T) -> Self {
+        Self {
+            tx: envelope.encoded_2718().into(),
+            max_block_number: None,
+            preferences: PrivateTransactionPreferences {
+                fast: None,
+                validity: None,
+                privacy: None,
+            },
```

Comment:
can we use a new function for this as well, from functions alone are more obfuscated

Reply:
is empty Bytes allowed?

Reply:
no but all those fields are pub,

the only thing missing here is this from fn as 

```
impl PrivateTransactionRequest {

pub fn new<T>() -> Self {
Self {
            tx: envelope.encoded_2718().into(),
            max_block_number: None,
            preferences: PrivateTransactionPreferences {
                fast: None,
                validity: None,
                privacy: None,
            },
}
```

and then the from function would just call  `PrivateTransactionRequest::new(tx)`

Reply:
makes some sense

Reply:
@mattsse I tried to impl TryFrom in order to return custom error but stuck with conflict. Is error needed?